src/storage/storage_structure/lists/lists.cpp

```diff
@@ -157,17 +160,31 @@ void Lists::fillInMemListsFromFrame(InMemList& inMemList, const uint8_t* frame,
             NullMask::copyNullMask(nullBufferInPage, elemPosInPage, inMemList.getNullMask(),
                 nextPosToWriteToInMemList, numElementsToReadInCurPage);
         }
+        readPropertyUpdatesToInMemListIfExists(inMemList, updatedPersistentListOffsets,
+            numElementsRead, numElementsToReadInCurPage, nextPosToWriteToInMemList);
         nextPosToWriteToInMemList += numElementsToReadInCurPage;
     } else {
         // If we have some deleted rels, we should check whether each rel has been deleted
         // or not before copying to inMemList.
         for (auto i = 0u; i < numElementsToReadInCurPage; i++) {
             auto relOffsetInList = numElementsRead + i;
             if (!deletedRelOffsetsInList.contains(relOffsetInList)) {
-                memcpy(listData, frameData, elementSize);
-                if (inMemList.hasNullBuffer()) {
-                    NullMask::copyNullMask(nullBufferInPage, elemPosInPage, inMemList.getNullMask(),
-                        nextPosToWriteToInMemList, 1 /* numBitsToCopy */);
+                if (updatedPersistentListOffsets != nullptr &&
+                    updatedPersistentListOffsets->listOffsetFTIdxMap.contains(relOffsetInList)) {
+                    // If the current rel has updates, we should read from the updateStore.
+                    listsUpdatesStore->readPropertyUpdatesToInMemList(
```

Comment:
I think this function should be singular, right? You are reading a single update to InMemList. So let's call it "readPropertyUpdateToInMemList".

Reply:
renamed to `readPropertyUpdateToInMemList `